src/components/service/ServiceCard.tsx

```diff
@@ -1,11 +1,11 @@
 
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import ServiceError from './ServiceError';
 import ServiceInfo from './ServiceInfo';
 import StorageQuery from './StorageQuery';
 import PreimageQuery from './PreimageQuery';
 import LookupHistoryQuery from './LookupHistoryQuery';
-import { getServiceChangeType } from './serviceUtils';
+import { getServiceChangeType, discoverStorageKeysForService, discoverPreimageKeysForService, discoverLookupHistoryKeysForService } from './serviceUtils';
```

Comment:
Seems that the `discoverLookupHistoryKeysForService` function does not filter-out entries that are found in preimage/storage.

We do that in the UI, but because of that the count does not match the displayed results. Can you update the function to actually filter out false-positives and remove the UI-only filtering logic?